### DISCUSSION TITLE
Request for a New Component: Spiral Waveguide Supporting n Parallel Waveguides

### DISCUSSION THREAD

**Discussion by Nianjia-Yang** [gdsfactory/gdsfactory#3491] [Q&A]
Hello, and thank you for developing GDSFactory — an incredibly powerful tool for photonic design!

Recently, I have been working on designing a spiral waveguide structure with n parallel waveguides that are evenly spaced and spiral together. While I was able to achieve this design using Boolean operations to combine multiple single-waveguide spirals, this approach has certain limitations in terms of efficiency and layout integration.

I would like to propose adding a new built-in component to GDSFactory that can directly generate such a multi-waveguide spiral structure. Ideally, this component would support the following features:

1. Support for configuring the number of parallel waveguides and their relative spacing.

2. Compatibility with various waveguide cross-sections and widths.

3. Automated alignment of input and output ports.

Introducing this component would significantly improve design efficiency and better facilitate the creation of complex photonic circuits.

Below is an example of the device I tried to create . If needed, I’d be happy to share my code and the implementation details for reference.

Looking forward to your feedback and consideration!
![image](https://github.com/user-attachments/assets/21469e4b-8b78-4073-856c-f43c1473cb47)
![image](https://github.com/user-attachments/assets/6fc2c0ea-0211-4ef9-9d9e-88c98882683c)

**Comment by joamatab** [comment]
Thank you for your kind words and for sharing your ideas about enhancing GDSFactory! Your proposal for a multi-waveguide spiral structure sounds fantastic and could be a valuable addition to the tool.

Please share the code and implementation details, and we’ll explore integrating it into GDSFactory.

Looking forward to collaborating with you!

Best regards,
Joaquin

**Reply by Nianjia-Yang** [reply]
Hi Joaquin,

Thank you so much for your kind words and for taking the time to respond. It’s incredibly encouraging to see such openness to new ideas! Below, I’ve shared the code and additional implementation details:
```
import gdsfactory as gf

c = gf.Component()
w1 = gf.cross_section.strip(width=25,radius = 600)
w2 = gf.cross_section.strip(width=15,radius = 600)
s1 =c << gf.components.spiral(length=1000,
                         bend='bend_euler',
                         straight='straight',
                         cross_section=w1,
                         spacing=40,
                         n_loops=6
                         )
s2 =c << gf.components.spiral(length=1000,
                         bend='bend_euler',
                         straight='straight',
                         cross_section=w2,
                         spacing=40,
                         n_loops=6
                         )
c = gf.boolean(s1, s2, operation="xor")
c.draw_ports()
c.show()
```
```
import gdsfactory as gf

c = gf.Component()
w1 = gf.cross_section.strip(width=25,)
w2 = gf.cross_section.strip(width=15,)
s1 =c << gf.components.spiral_double(
                         min_bend_radius = 600,
                         cross_section=w1,
                         separation=40,
                         number_of_loops=6
                         )
s2 =c << gf.components.spiral_double(
                         min_bend_radius = 600,
                         cross_section=w2,
                         separation=40,
                         number_of_loops=6
                         )
c = gf.boolean(s1, s2, operation="xor")
c.show()
```
This is just a simple idea and starting point I’ve put together. However, as a beginner with limited skills, I know there’s room for improvement. I hope you can refine and enhance the code further to make this feature more robust.

The key challenge I faced was using Boolean operations to merge multiple spirals, which made the process less efficient and harder to integrate. A dedicated component would greatly improve the workflow and enhance the design’s scalability for photonic integrated circuits that require compact, parallel routing.

Thank you again for all the hard work on GDSFactory. I hope this tool continues to grow and get better with contributions like these!

Best regards,
Nianjia
